File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int countLetters(string text);
int countWords(string text);
int countSentences(string text);
int findGrade(int words, int letters, int sentences);


int main(void){
    string text = get_string("Text: ");

    int words = countWords(text);
    int letters = countLetters(text);
    int sentences = countSentences(text);
    int grade = findGrade(words, letters, sentences);

    if (grade < 1){
        printf("Before Grade 1\n");
    } else if (grade >= 16){
        printf("Grade 16+\n");
    } else {
        printf("Grade %i\n", grade);
    }

}

int countLetters(string text){
    int letters = 0;

    for(int i = 0, n = strlen(text); i < n; i++){
        if(isalnum(text[i])){
            letters++;
        }
    }

    return letters;
}

int countWords(string text){
    int words = 1; //considering that the last word doesn't have a space, it just has a end dot, we should start the count in 1

    for(int i = 0, n = strlen(text); i < n; i++){
        if(isspace(text[i])){
            words++;
        }
    }

    return words;
}

int countSentences(string text){
    int sentences = 0; //considering that the last word doesn't have a space, it just has a end dot, we should start the count in 1

    for(int i = 0, n = strlen(text); i < n; i++){
        if((text[i] == 33) || (text[i] == 46) || (text[i] == 63)){
            sentences++;
        }
    }

    return sentences;
}

int findGrade(int words, int letters, int sentences){
    //applying the algorithm
    float l = (100 * (float) letters) / (float) words;
    float s = (100 * (float) sentences) / (float) words;

    float index = (0.0588 * l) - (0.296 * s) - 15.8;

    return round(index);
}

File: pset6/readability/readability.py
from cs50 import get_string

def main():
    text = get_string('Text: ')

    l = countLetters(text)
    w = countWords(text)
    s = countSentences(text)

    grade = colemanliau(l, w, s)
    if(grade < 1):
        print('Before Grade 1')
    elif(grade >= 16):
        print('Grade 16+')
    else:
        print(f'Grade {grade}')

#functions to count words/letters/sentences
def countLetters(text):
    letters = 0
    for i in range(len(text)):
        if(text[i].isalpha()):
            letters += 1
        else:
            continue
    return letters

def countWords(text):
    words = 1
    for i in range(len(text)):
        if(text[i].isspace()):
            words += 1
        else:
            continue
    return words

def countSentences(text):
    sentences = 0
    for i in range(len(text)):
        if(('.' in text[i]) or ('!' in text[i]) or ('?' in text[i])):
            sentences += 1
        else:
            continue
    return sentences

#average function
def average(value, words):
   return (value * 100) / words


#Coleman Liau formula
def colemanliau(l, w, s):
    L = average(l, w)
    S = average(s, w)

    return round(0.0588 * L - 0.296 * S - 15.8)


main()

